src/axom/quest/examples/quest_marching_cubes_example.cpp

```diff
@@ -0,0 +1,1019 @@
+
+    mca.set_function_field(function_name());
+
+    sidre::DataStore objectDS;
+    sidre::Group* meshGroup = objectDS.getRoot()->createGroup(name() + "_mesh");
+    axom::mint::UnstructuredMesh<axom::mint::SINGLE_SHAPE> surfaceMesh(
+      DIM,
+      DIM == 2 ? mint::CellType::SEGMENT : mint::CellType::TRIANGLE,
+      meshGroup);
+    mca.set_output_mesh(&surfaceMesh);
+
+    axom::utilities::Timer computeTimer(false);
+#ifdef AXOM_USE_MPI
+    MPI_Barrier(MPI_COMM_WORLD);
+#endif
+    computeTimer.start();
+    mca.compute_iso_surface(params.contourVal);
+    computeTimer.stop();
+    print_timing_stats(computeTimer, name() + " contour");
+
+    add_rank_offset_to_surface_mesh_domain_ids(computationalMesh.domain_count(),
+                                               surfaceMesh);
+
+    int localErrCount = 0;
+    if(params.checkResults)
+    {
+      localErrCount +=
+        check_contour_surface(surfaceMesh, params.contourVal, "diff");
+    }
+
+    save_mesh(*meshGroup, name() + "_surface_mesh");
+    SLIC_INFO(
+      axom::fmt::format("Wrote {} contour in {}_surface_mesh", name(), name()));
+
+    return localErrCount;
+  }
+
+  void compute_nodal_distance(BlueprintStructuredMesh& bpMesh)
+  {
+    SLIC_ASSERT(bpMesh.dimension() == DIM);
+    conduit::Node& bpNode = bpMesh.as_conduit_node();
+    for(conduit::Node& dom : bpNode.children())
+    {
+      // Access the coordinates
+      auto cellCounts = bpMesh.domain_lengths(dom);
+      axom::StackArray<axom::IndexType, DIM> shape;
+      for(int i = 0; i < DIM; ++i)
+      {
+        shape[i] = 1 + cellCounts[i];
+      }
+      conduit::index_t pointCount = bpMesh.node_count(dom);
+      conduit::Node& coordsValues =
+        dom.fetch_existing("coordsets/coords/values");
+      double* coordsPtrs[DIM];
+      axom::ArrayView<double, DIM> coordsViews[DIM];
+      for(int d = 0; d < DIM; ++d)
+      {
+        coordsPtrs[d] = coordsValues[d].as_double_ptr();
+        coordsViews[d] = axom::ArrayView<double, DIM>(coordsPtrs[d], shape);
+      }
+      axom::ArrayView<axom::primal::Point<double, DIM>, DIM> coordsView(
+        (axom::primal::Point<double, DIM>*)coordsValues.data_ptr(),
+        shape);
+
+      // Create the nodal function data.
+      conduit::Node& fieldNode = dom["fields"][function_name()];
+      fieldNode["association"] = "vertex";
+      fieldNode["topology"] = "mesh";
+      fieldNode["volume_dependent"] = "false";
+      fieldNode["values"].set(conduit::DataType::float64(pointCount));
+      double* d = fieldNode["values"].value();
+      axom::ArrayView<double, DIM> fieldView(d, shape);
+
+      // Set the nodal data to the distance from center.
+      for(int i = 0; i < pointCount; ++i)
+      {
+        axom::primal::Point<double, DIM> pt;
+        for(int d = 0; d < DIM; ++d)
+        {
+          pt[d] = coordsViews[d].flatIndex(i);
+        };
+        fieldView.flatIndex(i) = value(pt);
```

Comment:
A comment that value is the virtual function defining the distance in a derived class would help here!

Since we're only looking at two cases (plane and sphere), a (local) lambda that takes a point and returns a distance would likely be simpler and easier to understand.

Reply:
For the lambdas, do you mean something like this?
` auto pointByComponents = [=](double x, double y){return axom::primal::Point<double, DIM>{x, y};};`
with an `{x,y,z]` version as well.